api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -132,6 +132,12 @@ public StructType partitionType() {
           for (PartitionField field : fields) {
             Type sourceType = schema.findType(field.sourceId());
             Type resultType = field.transform().getResultType(sourceType);
+
+            // When the source field has been dropped we cannot determine the type
+            if (resultType == null) {
```

Comment:
This is the only change i'm not sure of, What's the value of not leaving this null? To me unknown specifies that the type could (and will be set) in the future but in this case the partition spec is essentially now dead and will never have a valid response?

I don't know if it matters really that much but I was wondering what the motivation was for the change here?

Reply:
It cannot be `null` because that will raise an exception when constructing the `StructType` below: `Types.StructType.of(structFields);`. Just filtering is not allowed, since for V1 we have to maintain the position of the fields, therefore injecting a `NullType` feels to me the safest approach. Since the field is not part of the active partition spec, it won't be used anymore, so I think we're good.

Reply:
@Fokko actually should the `if` condition be `if (sourceType == null)`?

`getResultType` doesn't return null. e.g. `Bucket` transform
```
  public Type getResultType(Type sourceType) {
    return Types.IntegerType.get();
  }
```